### PR TITLE
Some housekeeping (update Sol, extend charid, force Wheaty to fail, add setting to disable char deletion)

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Some bits:
- Update sol, update build flags, change usage of `safe_script_file` to not log/crash
- Some charid changes from Eden
- Make sure the main body of Wheaty will always hard exit if there's a problem
- Add setting to disable character deletion

## Steps to test these changes

The game starts up and runs normally
You can create/delete characters, depending on your settings
You can turn off character deletion
You can hot-reload script, with and without errors in them, and the server will stay up
You can force a crash from Windows, and it'll generate a proper Wheaty report

### CODE REVIEW

**Comment by zach2good** [comment]
I'm going to be away in London for a week, so feel free to add to or merge this branch as you others see fit

File: src/map/lua/lua_baseentity.cpp [hunk 1]
```diff
@@ -13239,25 +13267,25 @@ void CLuaBaseEntity::castSpell(sol::object const& spell, sol::object entity)
...
                 PEntity->PAI->Cast(static_cast<CMobEntity*>(PEntity)->GetBattleTargetID(), spellid);
-            }
-        }));
+            } }));
     }
     else
     {
-        m_PBaseEntity->PAI->QueueAction(queueAction_t(0ms, true, [](auto PEntity) {
+        m_PBaseEntity->PAI->QueueAction(queueAction_t(0ms, true, [](auto PEntity)
+                                                      {
```
Comment (TeoTwawki) [inline src/map/lua/lua_baseentity.cpp]: whitespace had an accident here.

File: src/map/status_effect_container.cpp [hunk 2]
```diff
@@ -1751,7 +1753,8 @@ void CStatusEffectContainer::HandleAura(CStatusEffect* PStatusEffect)
     {
         if (auraTarget == AURA_TARGET::ALLIES)
         {
-            PEntity->ForParty([&](CBattleEntity* PMember) {
+            PEntity->ForParty([&](CBattleEntity* PMember)
+                              {
```
Comment (TeoTwawki) [inline src/map/status_effect_container.cpp]: here too

File: src/map/utils/battleutils.cpp [hunk 3]
```diff
@@ -1046,11 +1046,11 @@ namespace battleutils
             }
             else if (PAttacker->objtype == TYPE_TRUST && PAttacker->PMaster)
             {
-                static_cast<CCharEntity*>(PAttacker->PMaster)->ForPartyWithTrusts([&](CBattleEntity* PMember) {
+                static_cast<CCharEntity*>(PAttacker->PMaster)->ForPartyWithTrusts([&](CBattleEntity* PMember)
+                                                                                  {
```
Comment (TeoTwawki) [inline src/map/utils/battleutils.cpp]: seems to be being auto aligned to the square brackets

File: src/map/utils/zoneutils.cpp [hunk 4]
```diff
@@ -333,7 +337,15 @@ namespace zoneutils
         }
 
         // handle npc spawn functions after they're all done loading
-        ForEachZone([](CZone* PZone) { PZone->ForEachNpc([](CNpcEntity* PNpc) { luautils::OnNpcSpawn(PNpc); }); });
+        // clang-format off
+        ForEachZone([](CZone* PZone)
```
Comment (TeoTwawki) [inline src/map/utils/zoneutils.cpp:318]: this type of this was the sole exception to brace on newline in the style guide. if we're dropping that the guide will need an update.

Comment (Xaver-DaRed) [reply src/map/utils/zoneutils.cpp:318]: I'll fix the others you pointed out. This one, I'm not sure of, so we'll have to wait on Zach's/Clay's imput

Comment (claywar) [reply src/map/utils/zoneutils.cpp:318]: clang-format chews up Lambdas in a nasty way.  Need to make sure the ones you fix are wrapped in // clang-format off/on blocks

**Comment by zach2good** [comment]
Updated and ready for review

**Review by claywar — APPROVED** [review]

**Pull request merged**